pkg/test/e2e/cilium/cilium_test.go

```diff
@@ -0,0 +1,460 @@
+				}
+				for _, c := range pod.Status.Conditions {
+					if c.Type == corev1.PodReady {
+						if c.Status != corev1.ConditionTrue {
+							podHealthy = false
+							t.Log("not ready", pod.Name, c.Type, c.Status)
+						}
+					} else if c.Type == corev1.ContainersReady {
+						if c.Status != corev1.ConditionTrue {
+							podHealthy = false
+							t.Log("not container ready", pod.Name, c.Type, c.Status)
+						}
+					}
+				}
+
+				if !podHealthy {
+					t.Logf("%q not healthy", pod.Name)
+				}
+
+				allHealthy = allHealthy && podHealthy
+			}
+
+			if !allHealthy {
+				t.Logf("not all pods healthy yet...")
+				return false, nil
+			}
+
+			t.Logf("all pods healthy")
+
+			return true, nil
+		})
+		if err != nil {
+			t.Fatalf("pods never became ready: %v", err)
+		}
+	}
+}
+
+func filterByName(pods []corev1.Pod, names []string) []corev1.Pod {
+	c := 0
+	for _, p := range pods {
+		for _, name := range names {
+			if p.Labels["name"] == name {
+				c++
+				pods = append(pods, p)
+			}
+		}
+	}
+	return pods[len(pods)-c:]
+}
+
+func makeScheme() *runtime.Scheme {
+	var s = runtime.NewScheme()
+	_ = serializer.NewCodecFactory(s)
+	_ = runtime.NewParameterCodec(s)
+	utilruntime.Must(appsv1.AddToScheme(s))
+	utilruntime.Must(corev1.AddToScheme(s))
+	utilruntime.Must(ciliumv2.AddToScheme(s))
+	return s
+}
+
+func resourcesFromYaml(s *runtime.Scheme) ([]runtime.Object, error) {
+	data, err := ioutil.ReadFile("./testdata/connectivity-check.yaml")
+	if err != nil {
+		return nil, err
+	}
+
+	manifests, err := yamlutil.ParseMultipleDocuments(bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+
+	sr := kjson.NewSerializerWithOptions(&kjson.SimpleMetaFactory{}, s, s, kjson.SerializerOptions{})
+
+	var objs []runtime.Object
+	for _, m := range manifests {
+		obj, err := runtime.Decode(sr, m.Raw)
+		if err != nil {
+			return nil, err
+		}
+		objs = append(objs, obj)
+	}
+
+	return objs, nil
+}
+
+// creates a usercluster on aws
+func createUsercluster(t *testing.T) (string, string, func(), error) {
+	var teardowns []func()
+	cleanup := func() {
+		n := len(teardowns)
+		for i := range teardowns {
+			teardowns[n-1-i]()
+		}
+	} // get kubermatic-api client
+	token, err := utils.RetrieveMasterToken(context.Background())
+	if err != nil {
+		return "", "", nil, err
+	}
+
+	apicli := utils.NewTestClient(token, t)
+
+	// create a project
+	project, err := apicli.CreateProject(projectName)
+	if err != nil {
+		return "", "", nil, err
+	}
+	teardowns = append(teardowns, func() {
+		err := apicli.DeleteProject(project.ID)
+		if err != nil {
+			t.Errorf("failed to delete project %s: %s", project.ID, err)
+		}
+	})
+
+	// create a usercluster on aws
+	cluster, err := apicli.CreateAWSCluster(project.ID, seed, userclusterName,
+		secretAccessKey, accessKeyID, utils.KubernetesVersion(),
+		"aws-eu-central-1a", "eu-central-1a", 1, true, &models.CNIPluginSettings{
+			Version: "v1.11",
+			Type:    "cilium",
+		})
```

Comment:
Not 100% sure, but should we consider using ebpf proxy mode for the cluster?

Reply:
Actually both ebpf and non-ebpf variant would be interesting..

Reply:
Yep. Let’s add.